freedb/protocols.md

```diff
@@ -1,8 +1,311 @@
+and headers:
+
+| Parameter Name | Value                     |
+| -------------- | ------------------------- |
+| `sheet`        | `<sheet_name>`            |
+| `tqx`          | `responseHandler:freeleh` |
+| `tq`           | `<gsheet_query>`          |
+| `headers`      | `1`                       |
+
+| Header Name   | Value                 |
+| ------------- | --------------------- |
+| Content-Type  | `application/json`    |
+| Authorization | `Bearer <auth_token>` |
+
+Replace `<sheet_name>` with the sheet name that we're going to operate on, `<gsheet_query>` with the query that we
+want to run, and `<auth_token>` with the token that you use to call the Google Sheet V4 API.
+
+For example, suppose you want to run the "SELECT A" query on `Sheet1` of a spreadsheet with ID `ABCD` with `TOKEN` as
+the authorization token the actual request will look like this:
+```
+GET /spreadsheets/d/ABCD/gviz/tq?sheet=Sheet1&tqx=responseHandler:freeleh&tq=SELECT%20A&headers=1 HTTP/2
+Host: docs.google.com
+Content-Type: application/json
+Authorization: Bearer TOKEN
+```
+
+If everything works well, you will get a response like this:
+```
+/*O_o*/
+freeleh({"version":"0.6","reqId":"0","status":"ok","sig":"880399325","table":{"cols":[{"id":"B","label":"name","type":"string"},{"id":"C","label":"age","type":"number","pattern":"General"},{"id":"D","label":"date of birth","type":"date","pattern":"m-d-yyyy"}],"rows":[{"c":[{"v":"name1"},{"v":10.0,"f":"10"},{"v":"Date(1999,0,1)","f":"1-1-1999"}]},{"c":[{"v":"name2"},{"v":11.0,"f":"11"},{"v":"Date(2000,0,1)","f":"1-1-2000"}]},{"c":[{"v":"name3"},{"v":12.0,"f":"12"},{"v":"Date(2001,0,1)","f":"1-1-2001"}]}],"parsedNumHeaders":0}});
+```
+
+Before we can process the response, we need to discard any character that comes before the first `{` character and
+those that come after the last `}` character, after we discard those characters we will get a valid JSON string.
+
+```
+{"version":"0.6","reqId":"0","status":"ok","sig":"880399325","table":{"cols":[{"id":"B","label":"name","type":"string"},{"id":"C","label":"age","type":"number","pattern":"General"},{"id":"D","label":"date of birth","type":"date","pattern":"m-d-yyyy"}],"rows":[{"c":[{"v":"name1"},{"v":10.0,"f":"10"},{"v":"Date(1999,0,1)","f":"1-1-1999"}]},{"c":[{"v":"name2"},{"v":11.0,"f":"11"},{"v":"Date(2000,0,1)","f":"1-1-2000"}]},{"c":[{"v":"name3"},{"v":12.0,"f":"12"},{"v":"Date(2001,0,1)","f":"1-1-2001"}]}],"parsedNumHeaders":0}}
+```
+
+`.table.cols` contain the list of column info objects that we included in the `SELECT` clause in the same order
+of appearance.
+
+`.table.rows` contain the list of row objects. The value of a `i`-th row can be accessed via `.table.rows[i].c` key.
+Inside of the row value object you will find `c` and `f` key, where `c` corresponds to the formatted value while `f`
+corresponds to the raw value.
+
+To parse the value, use `f` value if possible and fallback to `v` value and convert the value to your language native
```

Comment:
I think this part requires more information about why we have these distinctions and what kind of conversion is required. Otherwise people might be confused why we have to care about `c` and `f` at all.

Reply:
Updated to the latest logic. Also added how to convert the value.